src/runtime_src/core/pcie/emulation/cpu_em/generic_pcie_hal2/shim.cxx

```diff
@@ -1715,22 +1715,22 @@ namespace xclcpuemhal2
 
     DEBUG_MSGS("%s, %d( src_boHandle: %x dst_boHandle: %x size: %zx  dst_offset: %zx src_offset: %zx )\n", __func__, __LINE__, src_boHandle, dst_boHandle, size, dst_offset, src_offset);
 
-    xclemulation::drm_xocl_bo *sBO = xclGetBoByHandle(src_boHandle);
+    auto sBO = xclGetBoByHandle(src_boHandle);
     if (!sBO)
     {
       PRINTENDFUNC;
       return -1;
     }
 
-    xclemulation::drm_xocl_bo *dBO = xclGetBoByHandle(dst_boHandle);
+    auto dBO = xclGetBoByHandle(dst_boHandle);
     if (!dBO)
     {
       PRINTENDFUNC;
       return -1;
     }
 
     // source buffer is host_only and destination buffer is device_only
-    if (xclemulation::xocl_bo_host_only(sBO) && !xclemulation::xocl_bo_p2p(sBO) && xclemulation::xocl_bo_dev_only(dBO))
+    if (xclemulation::xocl_bo_host_only(sBO.get()) && !xclemulation::xocl_bo_p2p(sBO.get()) && xclemulation::xocl_bo_dev_only(dBO.get()))
```

Comment:
can you move all of these functions also start using shared pointer?

Reply:
Agreed. The functions are using shared pointers now.